sdk/identity/azure-identity/src/chained_token_credential.cpp

```diff
@@ -66,10 +69,33 @@ ChainedTokenCredentialImpl::ChainedTokenCredentialImpl(
 AccessToken ChainedTokenCredentialImpl::GetToken(
     std::string const& credentialName,
     TokenRequestContext const& tokenRequestContext,
-    Context const& context) const
+    Context const& context)
 {
-  for (auto const& source : m_sources)
+  std::unique_lock<std::mutex> lock(m_cachingMutex, std::defer_lock);
+
+  constexpr size_t maxSentinel = std::numeric_limits<std::size_t>::max();
```

Comment:
1. `maxSentinel` => `MaxSentinel`, but
2. If we introduce `SuccessfulSourceNotSet` then this declaration is no longer necessary.